chainerx_cc/chainerx/cuda/cuda_device/linalg.cu

```diff
@@ -204,5 +284,142 @@ public:
 
 CHAINERX_CUDA_REGISTER_KERNEL(InverseKernel, CudaInverseKernel);
 
+class CudaSVDKernel : public SVDKernel {
+public:
+    std::tuple<Array, Array, Array> Call(const Array& a, bool full_matrices, bool compute_uv) override {
+        Device& device = a.device();
+        Dtype dtype = a.dtype();
+        CudaSetDeviceScope scope{device.index()};
+
+        CHAINERX_ASSERT(a.ndim() == 2);
+
+        int64_t n = a.shape()[0];
+        int64_t m = a.shape()[1];
+
+        Array x = Empty(Shape{n, m}, dtype, device);
+        bool trans_flag;
+
+        if (m >= n) {
+            device.backend().CallKernel<CopyKernel>(a, x);
+            trans_flag = false;
+        } else {
+            m = a.shape()[0];
+            n = a.shape()[1];
+            x = x.Reshape(Shape{n, m});
+            device.backend().CallKernel<CopyKernel>(a.Transpose(), x);
+            trans_flag = true;
+        }
+        int64_t mn = std::min(m, n);
+
+        Array u{};
+        Array vt{};
+
+        if (compute_uv) {
+            if (full_matrices) {
+                u = Empty(Shape{m, m}, dtype, device);
+                vt = Empty(Shape{n, n}, dtype, device);
+            } else {
+                u = Empty(Shape{mn, m}, dtype, device);
+                vt = Empty(Shape{mn, n}, dtype, device);
+            }
+        } else {
+            u = Empty(Shape{0}, dtype, device);
+            vt = Empty(Shape{0}, dtype, device);
```

Comment:
A minor suggestion but how about conditionally assign the shapes instead of the arrays and allocate `u` and `vt` using these conditional shapes e.g. before `Array s ...`, mostly to reduce the amount of code.